llvm/include/llvm/SYCL/KernelProperties.h

```diff
@@ -34,16 +34,28 @@ class KernelProperties {
   //
   // As of now, all arguments sharing the same memory bank share the 
   // same bundle.
+
+  enum struct MemoryType {
+      DEFAULT,
+      DDR,
+      HBM
```

Comment:
Curious these uppercase. It looks like macros. Use lowercase?
Replace `default` by non C++ keyword. `unspecified`? I am running out of creativity today...
Or just `Default`.

Reply:
Also above the comment `this is  the bundle` could be simplified as `this bundle`.